src/goto-instrument/accelerate/util.cpp

```diff
@@ -114,8 +116,10 @@ typet join_types(const typet &t1, const typet &t2)
     }
   }
 
-  std::cerr << "Failed to join types: "
-            << t1.pretty() << " and " << t2.pretty()
-            << '\n';
-  UNREACHABLE;
+#ifdef DEBUG
+  std::cout << "Failed to join types: "
+            << t1.pretty() << " and " << t2.pretty() << "\n";
+#endif
+
+  INVARIANT(false, "Failed to join types");
```

Comment:
Hmm, why not have the full error message in there (plus start with a lower-case letter)?

Reply:
The message argument to INVARIANT should be a string literal (according to invariant.h).

Reply:
`"failed to join types: "+t1.pretty()+" and "+t2.pretty()` should be such?

Reply:
Not according to my understanding of "literal". It needs to be known at compile time, so it cannot include function calls.

Reply:
The type of the parameter is `const std::string &` - the type checker will be pretty happy with a run-time constructed string?!